lib/token_util.py

```diff
@@ -77,56 +83,54 @@ def token_str_expired(token_str):
     return expired
 
 
-def simple_scramble(data):
+def simple_scramble(in_buf):
     """Undo the simple scramble of HTCondor
-    simply XOR with 0xdeadbeef
-    Using defaults.BINARY_ENCODING (latin-1) to have a 1-to-1 characted-byte correspondence
 
-    Source: https://github.com/CoffeaTeam/jhub/blob/master/charts/coffea-casa-jhub/files/hub/auth.py#L196-L235
+    simply XOR with 0xdeadbeef
+    Source: https://github.com/CoffeaTeam/coffea-casa/blob/master/charts/coffea-casa/files/hub-extra/auth.py
 
     Args:
         data(bytearray): binary string to be unscrambled
 
     Returns:
         bytearray: an HTCondor scrambled binary string
     """
-    outb = "".encode(defaults.BINARY_ENCODING)
     deadbeef = [0xDE, 0xAD, 0xBE, 0xEF]
-    ldata = len(data)
-    lbeef = len(deadbeef)
-    for i in range(ldata):
-        if sys.version_info[0] == 2:
-            datum = struct.unpack("B", data[i])[0]
-        else:
-            datum = data[i]
-        rslt = datum ^ deadbeef[i % lbeef]
-        b1 = struct.pack("H", rslt)[0]
-        outb += ("%c" % b1).encode(defaults.BINARY_ENCODING)
-    return outb
+    out_buf = b""
+    for idx in range(len(in_buf)):
+        scramble = in_buf[idx] ^ deadbeef[idx % 4]  # 4 = len(deadbeef)
```

Comment:
Would it be better to write `len(deadbeef)` instead of a static value `4`?

Reply:
since DEADBEEF will never change I did it for performance reasons, adding the comment for clarity.
Should I still use len()? In the end si just a few usec.